src/Telegram.Bot.Core/Requests/RequestBase.cs

```diff
@@ -11,11 +14,9 @@ namespace Telegram.Bot.Requests
     /// <typeparam name="TResponse">Type of result expected in result</typeparam>
     public abstract class RequestBase<TResponse> : IRequest<TResponse>
     {
-        /// <inheritdoc />
         public HttpMethod Method { get; }
-
-        /// <inheritdoc />
-        public string MethodName { get; protected set; }
+        public string MethodName { get; }
+        public ITelegramBotJsonConverter JsonConverter { get; set; }
```

Comment:
`JsonConverter` is required for request serialization, so it has to be provided as a parameter in `ToHttpContentAsync`

Reply:
It's also used in methods of `FileRequestBase` class so I decided to implement it as property

Reply:
It should be a property on `IRequest.ToHttpContentAsync`. It's used literally everywhere and in effect is required.